test/specs/controller.spec.js

```diff
@@ -24,4 +24,36 @@ describe('controller', function() {
     chart.update();
     expect(buildData).toBe(chart.data.datasets[0].data);
   });
+
+  it('should group 3 levels of data', function() {
+    const tree = [
+      {key: 10, a: 'a1', b: 'b1', c: 'c1'},
```

Comment:
The test would be more complete, if the `c` level would be `c1...c4`. So the value for path `a1.b1.c1` would be 10 etc.